app/server/api/ussd_api.py

```diff
@@ -31,19 +32,24 @@ def post(self):
         user_input = post_data.get('text')
         service_code = post_data.get('serviceCode')
 
+
         if phone_number:
             msisdn = proccess_phone_number(phone_number, 'KE')
-            user = User.query.filter_by(phone=msisdn).first()
+            user = User.query.execution_options(show_all=True).filter_by(phone=msisdn).first()
+            # api chains all inputs that came through with *
+            latest_input = user_input.split('*')[-1]
             # TODO(ussd): 'exit_not_registered' if no user
             if None in [user, msisdn, session_id]:
                 current_menu = UssdMenu.find_by_name('exit_invalid_request')
                 text = menu_display_text_in_lang(current_menu, user)
             else:
-                current_menu = KenyaUssdProcessor.process_request(session_id, user_input, user)
-                ussd_session = create_or_update_session(session_id, user, current_menu, user_input, service_code)
+                current_menu = KenyaUssdProcessor.process_request(session_id, latest_input, user)
+                ussd_session = create_or_update_session(session_id, user, current_menu, latest_input, service_code)
```

Comment:
or should i save the entire chain of `user_input` to the session instead of just the latest?

Reply:
what's the downside to saving the entire chain, but just using the latest? Added complexity without a real  justification

Reply:
Having now played around with it, I definitely think it'd be good to save the chain to help with diagnostics

Reply:
good idea, will do